### PR TITLE
URLs to the API docs updated (#2177)

### PR DESCRIPTION
**Pull request by danielfernau**
URLs  to missing API docs added (https://github.com/emberjs/guides/pull/2085)

### CODE REVIEW

File: source/configuring-ember/disabling-prototype-extensions.md [hunk 1]
```diff
@@ -147,7 +147,7 @@ fullNameDidChange: observer('fullName', function() {
 })
 ```
 
-Evented functions are annotated using `Ember.on()`:
+Evented functions are annotated using [`Ember.on()`](https://emberjs.com/api/ember/2.15/namespaces/Ember/methods/on?anchor=on):
```
Comment (acorncom) [inline source/configuring-ember/disabling-prototype-extensions.md]: Whoops, this seems to have gone the wrong way ... Mind fixing that? 😀

File: source/configuring-ember/handling-deprecations.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ Fortunately, Ember provides a way for projects to deal with deprecations in an o
 ## Filtering Deprecations
 
 When your project has a lot of deprecations, you can start by filtering out deprecations that do not have to be addressed right away.  You
-can use the [deprecation handlers](http://emberjs.com/api/classes/Ember.Debug.html#method_registerDeprecationHandler) API to check for what
+can use the [deprecation handlers](https://emberjs.com/api/ember/2.15/classes/Ember.Debug/methods/registerDeprecationHandler?anchor=registerDeprecationHandler) API to check for what
```
Comment (acorncom) [inline source/configuring-ember/handling-deprecations.md]: Same here

File: source/models/creating-updating-and-deleting-records.md [hunk 3]
```diff
@@ -27,7 +27,7 @@ this.get('store').findRecord('person', 1).then(function(tyrion) {
 
 All of the Ember.js conveniences are available for
 modifying attributes. For example, you can use `Ember.Object`'s
-[`incrementProperty`](http://emberjs.com/api/classes/Ember.Object.html#method_incrementProperty) helper:
+[`incrementProperty`](https://emberjs.com/api/ember/2.15/classes/Ember.Object/methods/incrementProperty?anchor=incrementProperty) helper:
```
Comment (acorncom) [inline source/models/creating-updating-and-deleting-records.md]: And here

File: source/models/finding-records.md [hunk 4]
```diff
@@ -32,14 +32,14 @@ let blogPosts = this.get('store').peekAll('blog-post'); // => no network request
 
 `store.findAll()` returns a `DS.PromiseArray` that fulfills to a `DS.RecordArray` and `store.peekAll` directly returns a `DS.RecordArray`.
 
-It's important to note that `DS.RecordArray` is not a JavaScript array, it's an object that implements [`Ember.Enumerable`](http://emberjs.com/api/classes/Ember.Enumerable.html).
+It's important to note that `DS.RecordArray` is not a JavaScript array, it's an object that implements [`Ember.Enumerable`](https://emberjs.com/api/ember/2.15/classes/Ember.Enumerable).
```
Comment (acorncom) [inline source/models/finding-records.md]: And this as well

File: source/object-model/enumerables.md [hunk 5]
```diff
@@ -1,6 +1,6 @@
 In Ember.js, an enumerable is any object that contains a number of child
 objects, and which allows you to work with those children using the
-[Ember.Enumerable](http://emberjs.com/api/classes/Ember.Enumerable.html) API. The most common
+[Ember.Enumerable](https://emberjs.com/api/ember/2.15/classes/Ember.Enumerable) API. The most common
```
Comment (acorncom) [inline source/object-model/enumerables.md]: And this

File: source/object-model/enumerables.md [hunk 6]
```diff
@@ -46,11 +46,11 @@ in an observable fashion, you should use `myArray.get('firstObject')` and
 
 In the rest of this guide, we'll explore some of the most common enumerable
 conveniences. For the full list, please see the [Ember.Enumerable API
-reference documentation.](http://emberjs.com/api/classes/Ember.Enumerable.html)
+reference documentation.](https://emberjs.com/api/ember/2.15/classes/Ember.Enumerable)
```
Comment (acorncom) [inline source/object-model/enumerables.md]: Hmm, this one doesn't exist at `release`, is that why some of these still reference 2.15?  @toddjordan any thoughts on whether we want to support those in the future?

**Comment by acorncom** [comment]
This PR fixes #2177

**Comment by danielfernau** [comment]
@acorncom Thanks for your feedback!

I used the 2.15 docs there because opening, for example, 
http://emberjs.com/api/classes/Ember.Debug.html#method_registerDeprecationHandler
in a web browser redirects to
https://emberjs.com/api/ember/2.15/classes/Ember.Debug/methods/registerDeprecationHandler?anchor=registerDeprecationHandler

For now, I'll wait until we know the opinion of the other two learning team managers you mentioned...
Just let me know if I need to change anything!

**Comment by jenweber** [comment]
@acorncom I agree - we can fix the older links later. It's a big improvement!

I'd like to hear from Todd as well before merging.

Daniel, thanks so much for taking this on :D

File: source/configuring-ember/disabling-prototype-extensions.md (hunk 1 above)
Comment (sivakumar-kailasam) [reply source/configuring-ember/disabling-prototype-extensions.md]: This one doesn't exist on `release`

File: source/configuring-ember/handling-deprecations.md (hunk 2 above)
Comment (sivakumar-kailasam) [reply source/configuring-ember/handling-deprecations.md]: should remain on 2.15

File: source/models/creating-updating-and-deleting-records.md (hunk 3 above)
Comment (sivakumar-kailasam) [reply source/models/creating-updating-and-deleting-records.md]: should remain on 2.15

**Comment by danielfernau** [comment]
@jenweber You're welcome, I'm glad I was able to help :)

@sivakumar-kailasam Thanks for resolving the conflicts and merging my PR!

**Comment by sivakumar-kailasam** [comment]
@danielfernau We'll make sure to review things quicker the next time you send out a PR 😉

**Pull request merged**